micrometer-jersey2/build.gradle

```diff
@@ -0,0 +1,10 @@
+dependencies {
+    compile project(':micrometer-core')
+    compile 'org.glassfish.jersey.core:jersey-server:2.25.1'
```

Comment:
Shouldn't this be an optional dependency? I believe the typical pattern in Boot is to provide functionality guarded by ConditionalOnClass, but not express opinions on dependency versions except in starter poms.

Reply:
This "lib" provides standalone Jersey2 instrumentation support for micromter. The Spring-Boot auto-configuration is handled in `spring-boot-legacy` and `spring-boot` (2.x) respectively. Thanks for looking!

Reply:
Yep, we'll make it optional.

Reply:
I don't get it. If you don't pull in this lib (`micrometer-jersey2`) as a dependency in your project, the auto-configuration won't kick in as it does `@ConditionalOnClass(MicrometerApplicationEventListener.class)`. (Remember, this instrumentation support is not embedded in `micrometer-core`.)

Reply:
Presumably in a Jersey app you want instrumented, your app already has an opinion about the Jersey version it is using. Suppose you were using a version < 2.25.1. We don't want conflict resolution on a "latest version conflict resolution" build tool like Gradle to pull up your Jersey version simply because you wanted to instrument it.

Reply:
Ok, made it optional. Perhaps I am so used to maven and it's "nearest dependency wins" behaviour and I accept that apparently not every developer is checking his dependency tree.